lib/stdio/handle.js

```diff
@@ -1,43 +1,34 @@
-import {getStdioOptionType, isRegularUrl, isUnknownStdioString, isInputDirection} from './type.js';
+import {getStdioOptionType, isRegularUrl, isUnknownStdioString} from './type.js';
+import {addStreamDirection} from './direction.js';
 import {normalizeStdio} from './normalize.js';
 import {handleInputOption, handleInputFileOption} from './input.js';
 
 // Handle `input`, `inputFile`, `stdin`, `stdout` and `stderr` options, before spawning, in async/sync mode
 export const handleInput = (addProperties, options) => {
 	const stdio = normalizeStdio(options);
-	const stdioStreams = stdio.map((stdioOption, index) => getStdioStream(stdioOption, index, addProperties, options));
+	const stdioStreams = stdio
+		.map((stdioOption, index) => getStdioStream(stdioOption, index, options))
+		.map(stdioStream => addStreamDirection(stdioStream))
+		.map(stdioStream => addStreamProperties(stdioStream, addProperties));
```

Comment:
The main refactoring we are doing is to break down the `stdio` option normalization logic into three successive steps:
  1. Main step: retrieve type of value (stream, file path, etc.) and handle `input`/`inputFile` options
  2. Get the value's "direction"
  3. Add "stream properties", i.e. information that is both type-specific and direction-specific. For example, if the value is a file path and its direction is "input", we create a stream using `fs.createReadStream()`. Or, if the value is an iterable and its direction is "output", we fail since iterables are only allowed as inputs.

Breaking it into three steps is going to help implement #620 in the next PR.